bitcoind-rpc/src/main/scala/org/bitcoins/rpc/jsonmodels/BlockchainResult.scala

```diff
@@ -104,7 +105,25 @@ case class GetBlockHeaderResult(
     chainwork: String,
     previousblockhash: Option[DoubleSha256DigestBE],
     nextblockhash: Option[DoubleSha256DigestBE])
-    extends BlockchainResult
+    extends BlockchainResult {
+  def blockHeader(implicit cp: ChainParams): BlockHeader = {
+
+    val prevHash = {
+      if (height == 0 && previousblockhash.isEmpty) {
+        cp.genesisBlock.blockHeader.previousBlockHashBE
+      } else {
+        previousblockhash.get
+      }
+    }
+    BlockHeader(version = Int32(version),
+      //i think the only exception here is the genesis block? Why is this a option?
```

Comment:
This is the case, on another note i think we should get our `ChainApi` to return something similar that bitcoin core does (mimic the bitcoin core api) https://bitcoin.org/en/developer-reference#getblockheader

Reply:
Does this comment refer to line 106/115? Makes sense to me. Did some basic testing with CLI, and it appears so. It would be nice to get rid of the option, somehow... Could we do something _slightly bad_ like special-casing it to the empty block hash, to get rid of the option?

Reply:
The problem is figuring out what network that empty block hash is associated with, which is why I included the implicit chain param. But i think that is a bad solution to. I'm open to ideas.

Reply:
Looks like the prev block has is `DoubleSha256Digest.empty` regardless of the network? Looking at https://github.com/bitcoin-s/bitcoin-s-core/blob/100221e94321f96fdad8042a9d7ca2e6e9890688/core/src/main/scala/org/bitcoins/core/protocol/blockchain/ChainParams.scala#L150